### PR TITLE
Refactor FluidWarpMod

### PR DESCRIPTION
**Pull request by makotech222**
This is an attempt to fix issues with fluid warp suddenly failing to push fluids at seemingly random times. I've done quite a bit of refactoring to make things simpler. Some functionality has also changed.

- Split up the warp buildings into Providers/Requesters
- Logically, Providers now push out to requesters. This allows us to split the provider's mass evenly to all requesters. This fixes possibility that you have too many requesters for each providers, leaving some requesters with nothing, making it look like its not hooked up at all. This way, it behaves like regular pipes which split off to its consumers (right? I'm new to ONI, lol).
- Removed power requirement for now. Can add later?

One thing I was looking into was adding a 'Status' label on the gates that displays how many gates are connecting to the same channel. Could also add status label for when the gate can't push any fluid or something. Couldn't figure out how to add status labels though, could use help on this.

I've done some basic testing in debug mode. I'd like to get some more testing in longer games, which I will also do myself when I get time. There is some sub-optimal code when it comes to performance. I don't expect it to create any lag, but there can be improvements made if it does. Please report any issues here!

### CODE REVIEW

**Comment by Blindfold** [comment]
Nice! I really like that approach. I'll give it a test as soon as I can get the new mod system working on my Mac )

**Comment by makotech222** [comment]
Well from my testing, the mods been working perfectly now. Feel free to merge.

**Comment by Blindfold** [comment]
I can't make it work on my macBook. The mod crashes on load. Trying to debug now

File: Source/FluidWarpMod/WarpProviderBase.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    {
+        public const ConduitType CONDUIT_TYPE = FluidWarpMod_Utils.GAS_CONDUIT_PROVIDER_TYPE;
+        public const ConduitType _conduitType = ConduitType.Gas;
+        public const string ID = "WarpProviderGas";
+        public override BuildingDef CreateBuildingDef()
+        {
+
+            int width = 1;
+            int height = 2;
+            string anim = "valveliquid_kanim";
```
Comment (Blindfold) [inline Source/FluidWarpMod/WarpProviderBase.cs]: Could you please change it to "valvegas_kanim". Quite inconvenient to have gas gates which look like liquid gates.

File: Source/FluidWarpMod/LiquidWarpMod.cs [hunk 2]
```diff
@@ -10,29 +10,30 @@ internal class LiquidWarpMod_GeneratedBuildings_LoadGeneratedBuildings
...
 
-			ModUtil.AddBuildingToPlanScreen("Plumbing", LiquidWarpConfig.ID);
+            Strings.Add("STRINGS.BUILDINGS.PREFABS.WARPREQUESTERLIQUID.NAME", "Liquid Warp Requester");
+            Strings.Add("STRINGS.BUILDINGS.PREFABS.WARPREQUESTERLIQUID.DESC", "Liquid Warp Requesters provides an easy way to request liquids from a Provider");
+            Strings.Add("STRINGS.BUILDINGS.PREFABS.WARPREQUESTERLIQUID.EFFECT", "Place one liquid warp provider, and place one liquid warp requester. Synchronize your warpgates using the same channel.");
 
-			/*
-            List<string> category = (List<string>)TUNING.BUILDINGS.PLANORDER.First(po => po.category == new HashedString("Plumbing")).data;
-            category.Add(LiquidWarpConfig.ID);
-			*/
```
Comment (Blindfold) [inline Source/FluidWarpMod/LiquidWarpMod.cs]: Please cleanup the dead code.

File: Source/FluidWarpMod/GasWarpMod.cs [hunk 3]
```diff
@@ -9,30 +9,30 @@ internal class GasWarpMod_GeneratedBuildings_LoadGeneratedBuildings
...
 
-			ModUtil.AddBuildingToPlanScreen("HVAC", GasWarpConfig.ID);
+            Strings.Add("STRINGS.BUILDINGS.PREFABS.WARPREQUESTERGAS.NAME", "Gas Warp Requester");
+            Strings.Add("STRINGS.BUILDINGS.PREFABS.WARPREQUESTERGAS.DESC", "Gas Warp Requesters provides an easy way to request gases from a Provider.");
+            Strings.Add("STRINGS.BUILDINGS.PREFABS.WARPREQUESTERGAS.EFFECT", "Place one gas warp provider, and place one gas warp requester. Synchronize your warpgates using the same channel.");
 
-			/*
-            List<string> category = (List<string>)TUNING.BUILDINGS.PLANORDER.First(po => po.category == new HashedString("HVAC")).data;
-            category.Add(GasWarpConfig.ID);
-			*/
```
Comment (Blindfold) [inline Source/FluidWarpMod/GasWarpMod.cs]: To be removed

**Comment by makotech222** [comment]
Okay fixed the wrong animation.

Those other two comments are about code that is already removed.

**Comment by Blindfold** [comment]
Regarding two commented snippets - I mean that it is generally a good practice to delete code when you want to remove it, not comment it out. Just to keep the codebase clean.

Anyway... Good job! Thank you!

**Comment by javisar** [comment]
Sorry, @Blindfold and @makotech222 I've had this mod abandoned lately.
Thank you both for your efforts.
I both of you agree, I'll try to upload this mod to Steam this week.

**Pull request merged**